ember-server/shared/src/main/scala/org/http4s/ember/server/internal/StreamForking.scala

```diff
@@ -64,7 +64,7 @@ private[internal] object StreamForking {
           .handleErrorWith(stopFailed)
           .guarantee(available.release >> decrementRunning)
 
-        available.acquire >> incrementRunning >> F.start(fa).void
+        F.uncancelable(poll => available.acquire >> incrementRunning >> F.start(poll(fa)).void)
```

Comment:
So this is interesting :) actually I don't think that `poll` is needed, since the uncancelable scope does not extend to forked fibers. I wasn't sure about this so I tried an experiment:

```scala
 //> using lib "org.typelevel::cats-effect::3.4.1"

import cats.effect._

object App extends IOApp.Simple {

  def run = IO.uncancelable { poll =>
    IO.deferred[Unit].flatMap { gate =>
      (gate.complete(()) *> IO.never).start.flatMap { fiber =>
        gate.get *> fiber.cancel *> fiber.join.flatMap(IO.println(_))
      }
    }
  }

}
```

```
$ scala-cli test.scala 
Canceled()
```

Reply:
That is a very good point! It does actually makes sense that each fiber should have its own cancelation scope